client/src/includes/tabs.js

```diff
@@ -0,0 +1,339 @@
+        tabContent.hidden = false;
+      }
+
+      if (this.state.initialPageLoad) {
+        // On first load set the scroll to top to avoid scrolling to active section and header covering up tabs
+        setTimeout(() => {
+          window.scrollTo(0, 0);
+        }, this.state.transition * 2);
+      }
+
+      // Dispatch tab selected event for the rest of the admin to hook into if needed
+      // Trigger tab specific switch event
+      this.tabList.dispatchEvent(
+        new CustomEvent('switch', { detail: { tab: tab.dataset.tab } }),
+      );
+      // Dispatch tab-changed event on the document
+      document.dispatchEvent(new CustomEvent('tab-changed'));
+
+      // Set url hash and browser history
+      if (!this.disableURL) {
+        this.setURLHash(tabContentId);
+      }
+    }
+  }
+
+  /**
+   * Fade Up and In animation
+   * @param tabContent{HTMLElement}
+   */
+  animateIn(tabContent) {
+    setTimeout(() => {
+      // eslint-disable-next-line no-param-reassign
+      tabContent.hidden = false;
+      // Wait for hidden attribute to be applied then fade in
+      setTimeout(() => {
+        tabContent.classList.add(this.state.css.animate);
+      }, this.state.transition);
+    }, this.state.transition);
+  }
+
+  /**
+   * Fade Down and Out by removing css class
+   * @param tabContent{HTMLElement}
+   */
+  animateOut(tabContent) {
+    // Wait element to transition out and then hide with hidden
+    tabContent.classList.remove(this.state.css.animate);
+    setTimeout(() => {
+      // eslint-disable-next-line no-param-reassign
+      tabContent.hidden = true;
+    }, this.state.transition);
+  }
+
+  bindEvents() {
+    if (!this.tabButtons) {
+      return;
+    }
+
+    this.tabButtons.forEach((tab, index) => {
+      tab.addEventListener('click', (e) => {
+        e.preventDefault();
+        this.selectTab(tab);
+      });
+      tab.addEventListener('focus', () => {
+        this.selectTab(tab);
+      });
+      tab.addEventListener('keydown', this.keydownEventListener);
+      // Set index of tab used in keyboard controls
+      // eslint-disable-next-line no-param-reassign
+      tab.index = index;
+    });
+
+    // Select previous or next tab using history
+    window.addEventListener('popstate', (e) => {
+      if (e.state && e.state.tabContent) {
+        const tab = this.tabContainer.querySelector(
+          `a[href="#${e.state.tabContent}"][role="tab"]`,
+        );
+        if (tab) {
+          this.selectTab(tab);
+        }
+      }
+    });
+  }
+
+  /**
+   *  Handle keydown on tabs
+   * @param {Event}event
+   */
+  keydownEventListener(event) {
+    const keyPressed = event.key;
+    const { keys } = this.state;
+
+    switch (keyPressed) {
+      case keys.left:
+      case keys.right:
+        this.switchTabOnArrowPress(event);
+        break;
+      case keys.enter:
+      case keys.space:
+        event.preventDefault();
+        this.selectTab(event.target);
+        break;
```

Comment:
Isn’t this redundant with the change we’ve done to switch tabs on focus change?

Reply:
I guess so eh 😅